### ISSUE TITLE
Screenshots not saving on the emulator and hence library not working correctly!

### ISSUE THREAD

**Issue by GP2Engine** [Karumi/Shot#222]
### **Expected behaviour**
Run the "gradlew executeScreenshotTests -Precord" command to capture the initial UI state screenshots and then run the "gradlew executeScreenshotTests" and then receive the result of the test, regarding of whether the UI’s current state is any different than the initial state (w/ the report).

### **Actual behaviour**
When we run the "gradlew executeScreenshotTests -Precord" command, despite it gets executed correctly (and we receive successful messages like:

>  Screenshots recorded and saved at: C:\Users\User\AndroidStudioProjects\myapp\screenshots\debug\

> You can review the execution report here: C:\Users\User\AndroidStudioProjects\myapp\build\reports\shot\record\index.html

), 
it won't save the screenshots in our emulator's Download directory, either from the sdcard directory or from the emulated/0/ one. It does create the "screenshots-default" directory but it's empty. Although, in our project's /build/reports/shot directory, it does save the screenshots fine and it gives out reports that mention that "0 screenshot tests recorded." However, it also fails to retrieve the screenshots in the project's screenshots/debug directory (even with a manual DownloadScreenshots gradle command execution), as they're not saved in the emulator in the first place, as mentioned earlier. More specifically, we also get the below warnings during the execution of the "gradlew executeScreenshotTests -Precord" command:

> "Shot ADB warning: We could not pull screenshots from folder: /storage/emulated/0/Download/screenshots/com.example.myapp.test/screenshots-compose-default/

> Shot ADB warning: rm: /storage/emulated/0/Download/screenshots/com.example.myapp.test/screenshots-compose-default/: No such file or directory". 

Also, if we run "gradlew executeScreenshotTests" command (after the Preshot one), it will always pass, no matter what changes we do.
However, if we run our test manually, it successfully saves the screenshots both in the sdcard/download directory and inside the corresponding storage/emulated/0 one. Also in this case, we can see them in the screenshots/debug directory in our project (if we execute DownloadScreenshots command). Although, if we now attempt to execute the “executeScreenshotTests” command, it will just delete the screenshots and back to square one.

### **Steps to reproduce**
Just execute the “gradlew executeScreenshotTests -Precord” and “gradlew executeScreenshotTests” gradle commands with the below test. (as for the storage permissions, we made sure the app has the required storage permissions and we also set the android:requestLegacyExternalStorage="true" to both the App’s and the Test’s Manifest files.)

```
@get:Rule
val mRuntimePermissionRule: GrantPermissionRule = GrantPermissionRule.grant(
    Manifest.permission.READ_EXTERNAL_STORAGE,
    Manifest.permission.WRITE_EXTERNAL_STORAGE
)

@Test
fun testDemoActivity() {
    val activity = ActivityTestRule(
        SettingsActivity::class.java, false, false
    )
    activity.launchActivity(null)
    compareScreenshot(activity.activity)
}
```

### **Version of the library**
5.10.4
•	Used Gradle 6.5
•	Windows 10 64-bit

**Comment by SamCosta1** [comment]
I'm also seeing something very similar. Works on a real device but getting almost exactly the issue described on the emulator 😢

**Comment by AlmightyCZ** [comment]
Similar problem here, it's working fine on macOS, but same project on Windows Server 2016 Std is passing all the screenshot comparisons (even if an error is introduced) and not saving any screenshots. I can't test real device on the windows machine, so both cases are with emulator.

**Comment by pedrovgs** [comment]
Closed because it is duplicated by #140
